src/6model/reprs/NativeRef.c

```diff
@@ -597,6 +597,36 @@ void MVM_nativeref_write_lex_i(MVMThreadContext *tc, MVMObject *ref_obj, MVMint6
             break;
     }
 }
+void MVM_nativeref_write_lex_u(MVMThreadContext *tc, MVMObject *ref_obj, MVMuint64 value) {
+    MVMNativeRef *ref = (MVMNativeRef *)ref_obj;
+    MVMRegister *var = &(ref->body.u.lex.frame->env[ref->body.u.lex.env_idx]);
+    switch (ref->body.u.lex.type) {
+        case MVM_reg_uint8:
+            var->u8 = (MVMuint8)value;
+            break;
+        case MVM_reg_uint16:
+            var->u16 = (MVMuint16)value;
+            break;
+        case MVM_reg_uint32:
+            var->u32 = (MVMuint32)value;
+            break;
+        case MVM_reg_uint64:
+            var->u64 = value;
+            break;
+        case MVM_reg_int8:
```

Comment:
I think MVM_nativeref_write_lex_i need only handle signed int regs and MVM_nativeref_write_lex_u only unsigned int regs. native_ref_store_i and native_ref_store_u already check the ref type accordingly. By throwing in case of an unexpected reg type we can check whether my understanding is wrong or if we maybe take a wrong turn somewhere.

Reply:
Done in [fed7bb8](https://github.com/MoarVM/MoarVM/pull/1678/commits/fed7bb86ce28630083e48162d9574ed98212718e), no problems in NQP build+test or Rakudo build+test.